### ISSUE TITLE
AutoComplete, when in dialog, intercepts escape keypresses and click outside

### ISSUE THREAD

**Issue by mattcosta7** [primer/react#3003]
### Description

see https://github.com/primer/react/issues/2995 for some setup

When Autocomplete is used inside a dialog, `Escape` to close the dialog fails - seemingly because the AutoComplete overlay intercepts these keypresses before they get tot he containing dialog - even when the autocomplete is not currently focuses (so the overlay itself is not visible)

### Steps to reproduce

Autocomplete in a dialog, rendered in a deferred way to ensure that it orders ontop of the dialog content.

Attempt to close the dialog with escape - note the dialog is not closeable, and the handler is never called

### Version

v35.20.0

**Comment by mattcosta7** [comment]
cc @stephanieG0

**Comment by stephanieg0** [comment]
Just adding this here for visibility. 
I had to change some `playwright` tests that used `page.keyboard.press('Escape')` to instead do `page.click('[aria-label=Close]')`

https://github.com/github/github/pull/263278/commits/08caa7ce06659b9d8ea15c8c420b4338a753f0d7
https://github.com/github/github/pull/263278/commits/2438a66a2da80c5f61b43f1c54add4aff3310453

**Comment by green6erry** [comment]
@mperrotti Here's the original ticket. Thanks for looking into it!
